File: FrontEnd/src/models/attendeeEntity.ts
import {TextObservable} from "./observables/textObservable";
import {ConsentState} from "./consentState";

export class AttendeeEntity {

  id?: string;
  teamId: string;

  name: TextObservable;
  jobTitle: TextObservable;
  profilePictureUrl: TextObservable;
  photoConsent?: ConsentState;

  hasSignedInToday?: boolean;

  static fromJson(json: any): AttendeeEntity {
    return new AttendeeEntity(
      json.teamId,
      json.id,
      json.name,
      json.jobTitle,
      json.profilePictureUrl,
      json.photoConsent,
      json.hasSignedInToday);
  }

  toJson(): string {
    return JSON.stringify({
      id: this.id,
      teamId: this.teamId,
      name: this.name.value,
      jobTitle: this.jobTitle.value,
      profilePictureUrl: this.profilePictureUrl.value
    });
  }

  constructor(teamId: string,
              id?: string,
              name?: string,
              jobTitle?: string,
              profilePictureUrl?: string,
              photoConsent?: ConsentState,
              hasSignedInToday?: boolean) {
    this.id = id;
    this.teamId = teamId;
    this.name = new TextObservable(name);
    this.jobTitle = new TextObservable(jobTitle);
    this.profilePictureUrl = new TextObservable(profilePictureUrl);
    this.photoConsent = photoConsent;
    this.hasSignedInToday = hasSignedInToday;
  }
}

File: Backend/Models/AttendeeEntity.cs
using System;
using FunctionApp.Models;
using Newtonsoft.Json;

namespace Innovate.Models
{
    public record AttendeeEntity : EntityToDto
    {
        public string? Name { get; set; }
        public string? JobTitle { get; set; }
        public string? TeamId { get; set; }
        public string? ProfilePictureUrl { get; set; }

        public string? PhotoConsent { get; set; }
        public DateTimeOffset? LastCheckInDate { get; set; }

        public bool HasSignedInToday
        {
            get
            {
                if (LastCheckInDate == null)
                {
                    return false;
                }

                var start = NewZealandDateTimeOffsetHelper.GetStartOfTodayNzInUtc();
                var end = start.AddDays(1);
                return LastCheckInDate >= start && LastCheckInDate < end;
            }
        }
    }
}